enterprise/backend/test/metabase_enterprise/serialization/v2/e2e_test.clj

```diff
@@ -735,3 +738,25 @@
                     (is (thrown-with-msg? Exception #"Please upgrade"
                                           (cmd/v2-load! dump-dir {}))
                         "throws")))))))))))
+
+(deftest every-model-is-supported-test
+  (testing "Serialization support\n"
+    (testing "We know about every model"
+      (is (= (set (concat serdes.models/exported-models
+                          serdes.models/inlined-models
+                          serdes.models/excluded-models))
+             (set (map name (v2.entity-ids/toucan-models))))))
+
+    (let [should-have-entity-id (set (concat serdes.models/data-model serdes.models/content))]
+      (doseq [model (v2.entity-ids/toucan-models)]
+        (if (contains? should-have-entity-id (name model))
+          (testing (str "Model either has entity_id or a hash key: " (name model))
+            ;; `not=` is effectively `xor`
+            (is (not= (not= (get-method serdes/entity-id (name model))
+                            (get-method serdes/entity-id :default))
+                      (and (not= (get-method serdes/hash-fields model)
+                                 (get-method serdes/hash-fields :default))
+                           (serdes.backfill/has-entity-id? model)))))
+          (testing (str "Model shouldn't have entity_id defined: " (name model))
+            (is (= (get-method serdes/entity-id (name model))
```

Comment:
No `hash-fields` or `has-entity-id?` either.

Reply:
I've dropped check for them since there are few defining them: DashboardTab, DashboardCard, Dimension. Should I strip them from hash-fields/:hook/entity-id?

Also PulseChannel and PulseCard if I put them into `inline-models` (will see what's going on with them for real tomorrow).

Reply:
Plus DashboardTab and DashboardCard [depend on entity id](https://github.com/metabase/metabase/blob/master/src/metabase/models/dashboard.clj#L623) when loading, but I'm almost sure we can drop all of them and just load new ones, right?

Reply:
We prefer to keep the IDs of existing things where practical. So we should be fetching the current set of dashcards, building the dead/updated/new diff sets, and then `DELETE`ing, `INSERT`ing and UPDATE`ing them as necessary.